MINOR: Tag AWS instances with Jenkins build url

maxzheng opened the pull request:
This will allow us to trace leaked instances back to the job,
so that we can figure out what happened and fix the leak.

Testing: Verified Jenkins build URL is set on the AWS instances when running system tests. 
Backporting: Can this be backported to 0.10.2? The oldest branch we can backport the better.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

ijuma commented on Vagrantfile:
```diff
@@ -141,7 +141,10 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
-      aws.tags = { 'Name' => ec2_instance_name_prefix + "-" + Socket.gethostname + "-" + name }
+      aws.tags = {
+        'Name' => ec2_instance_name_prefix + "-" + Socket.gethostname + "-" + name,
+        'Jenkins Build' => ENV['BUILD_URL']
```
What naming convention do we want to use for tags? Would camel case be simpler perhaps?

maxzheng replied:
Sounds good. Did a search and it seems AWS also uses camel case in their examples: https://aws.amazon.com/premiumsupport/knowledge-center/ec2-resource-tags/

ijuma commented on Vagrantfile:
```diff
@@ -141,7 +141,10 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
-      aws.tags = { 'Name' => ec2_instance_name_prefix + "-" + Socket.gethostname + "-" + name }
+      aws.tags = {
+        'Name' => ec2_instance_name_prefix + "-" + Socket.gethostname + "-" + name,
+        'JenkinsBuild' => ENV['BUILD_URL']
```
Sorry, should it be `JenkinsBuildUrl`? Otherwise it could be interpreted as the build number/name or something.

maxzheng replied:
Sure. Updated -- sorry, had to recreate the commit as I accidentally merged stuff from another remote.

Commit:
MINOR: Tag AWS instances with Jenkins build url

This will allow us to trace leaked instances back to the job,
so that we can figure out what happened and fix the leak.

maxzheng commented:
Ping @ijuma : Can u merge and backport to 0.10.2? Thx

ijuma reviewed (approved):
LGTM, thanks.
